src/network/connection.ts

```diff
@@ -268,58 +313,69 @@ export class Connection extends EventEmitter {
-      request.callback(new NetworkError('Connection closed'), undefined)
-      return false
-    }
+    connectionsApiChannel.start.publish(request.diagnostic)
+
+    try {
+      if (this.#status !== ConnectionStatuses.CONNECTED) {
+        request.callback(new NetworkError('Connection closed'), undefined)
+        return false
+      }
 
-    let canWrite = true
+      let canWrite = true
 
-    const { correlationId, apiKey, apiVersion, payload: payloadFn, hasRequestHeaderTaggedFields } = request
+      const { correlationId, apiKey, apiVersion, payload: payloadFn, hasRequestHeaderTaggedFields } = request
 
-    const writer = Writer.create()
-      .appendInt16(apiKey)
-      .appendInt16(apiVersion)
-      .appendInt32(correlationId)
-      .appendString(this.#clientId, false)
+      const writer = Writer.create()
+        .appendInt16(apiKey)
+        .appendInt16(apiVersion)
+        .appendInt32(correlationId)
+        .appendString(this.#clientId, false)
 
-    if (hasRequestHeaderTaggedFields) {
-      writer.appendTaggedFields()
-    }
+      if (hasRequestHeaderTaggedFields) {
+        writer.appendTaggedFields()
+      }
 
-    const payload = payloadFn()
-    writer.appendFrom(payload)
-    writer.prependLength()
+      const payload = payloadFn()
+      writer.appendFrom(payload)
+      writer.prependLength()
 
-    // Write the header
-    this.#socket.cork()
+      // Write the header
+      this.#socket.cork()
 
-    if (!payload.context.noResponse) {
-      this.#inflightRequests.set(correlationId, request)
-    }
+      if (!payload.context.noResponse) {
+        this.#inflightRequests.set(correlationId, request)
+      }
 
-    /* c8 ignore next */
-    loggers.protocol({ apiKey: protocolAPIsById[apiKey], correlationId, request }, 'Sending request.')
+      /* c8 ignore next */
+      loggers.protocol({ apiKey: protocolAPIsById[apiKey], correlationId, request }, 'Sending request.')
 
-    for (const buf of writer.buffers) {
-      if (!this.#socket.write(buf)) {
-        canWrite = false
+      for (const buf of writer.buffers) {
+        if (!this.#socket.write(buf)) {
+          canWrite = false
+        }
       }
-    }
 
-    if (!canWrite) {
-      this.#socketMustBeDrained = true
-    }
+      if (!canWrite) {
+        this.#socketMustBeDrained = true
+      }
 
-    this.#socket.uncork()
+      this.#socket.uncork()
 
-    if (payload.context.noResponse) {
-      request.callback(null, canWrite)
-    }
+      if (payload.context.noResponse) {
+        request.callback(null, canWrite)
+      }
 
-    // debugDump(Date.now() % 100000, 'send', { owner: this.#ownerId, apiKey: protocolAPIsById[apiKey], correlationId })
+      // debugDump(Date.now() % 100000, 'send', { owner: this.#ownerId, apiKey: protocolAPIsById[apiKey], correlationId })
 
-    return canWrite
+      return canWrite
+      /* c8 ignore next 5 */
+    } catch (error) {
+      connectionsApiChannel.error.publish({ ...request.diagnostic, error })
```

Comment:
Same as other places, all related events should have reference equality.

Reply:
Thanks, I applied them everywhere! :)